app/commands/cart_services/prep_cart_for_payment.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+#
+# Copyright 2021 Victoria Garcia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class CartServices::PrepCartForPayment
+
+  MEMBERSHIP = CartItem::MEMBERSHIP
+
+  attr_reader :our_cart, :amount_to_charge, :good_to_go, :processing_cart
+
+  def initialize(cart_obj)
+    @original_cart_object = cart_obj
+    @transaction_bin = identify_transaction_bin(cart_obj)
+    @user = note_user(cart_obj)
+    @amount_to_charge = 0
+    @good_to_go = false
+    @initial_item_count = initial_purchasable_item_count(cart_obj)
+    @holdable_types_created = {reservation: false}
+  end
+
+  def call
+    prep_cart_items
+    if confirm_all_items_prepped
+      @good_to_go = true
+    else
+      @amount_to_charge = 0
+    end
+    {good_to_go: @good_to_go, amount_to_charge: @amount_to_charge, holdable_types_made: @holdable_types_created }
+  end
+
+  private
+
+  def identify_transaction_bin(cart_o)
+    cart_o.now_bin
+  end
+
+  def note_user(cart_o)
+    cart_o.now_bin.user
+  end
+
+  def initial_purchasable_item_count(cart_o)
+    cart_o.now_bin.cart_items.count
+  end
+
+  def prep_cart_items
+    @transaction_bin.reload
+
+    return if @transaction_bin.cart_items.blank?
+
+    ActiveRecord::Base.transaction(joinable: false, requires_new: true)  do
+      @transaction_bin.cart_items.each do |item|
+        prep_membership_item(item)
+      end
+    end
+  end
+
+  def prep_membership_item(cart_item)
```

Comment:
I gather that this module in general will need to be expanded if things other than `Membership` types end up in the cart?

Reply:
Actually, it looks like in general there's a membership assumption here. I know that's fine, because we talked about it, but I want the class structure to make it clear so that we don't see _generic_ cart classes that are membership-specific when we go to factor other ones out later.

Reply:
Yeah, that's why there are so many case statements where there's only one case.  Since I don't know what the other acquirables are going to look like, that seemed like the sanest way to put in a placeholder for them.  My hope is that they'll slot right in, and that maybe we won't have to do so much Acquirable type-checking, but it's hard to engineer for that when I don't know what, exactly, the situation with those is going to be.  I just tried to preserve flexibility at points where divergence in the needs of different kinds of acquirables might be expected.